frontend/js/src/settings/missing-data/MissingMBData.tsx

```diff
@@ -156,58 +174,45 @@ export default function MissingMBDataPage() {
   const handleClickPrevious = () => {
     if (currPage && currPage > 1) {
       setLoading(true);
-      const offset = (currPage - 1) * EXPECTED_ITEMS_PER_PAGE;
       const updatedPage = currPage - 1;
-      setMissingData(
-        missingDataProps.slice(offset - EXPECTED_ITEMS_PER_PAGE, offset) || []
-      );
-      setCurrPage(updatedPage);
+      setSearchParams({ page: updatedPage.toString() });
       afterDisplay();
-      window.history.pushState(null, "", `?page=${updatedPage}`);
     }
   };
 
   const handleClickNext = () => {
     if (currPage && currPage < totalPages) {
       setLoading(true);
-      const offset = currPage * EXPECTED_ITEMS_PER_PAGE;
       const updatedPage = currPage + 1;
-      setMissingData(
-        missingDataProps.slice(offset, offset + EXPECTED_ITEMS_PER_PAGE) || []
-      );
-      setCurrPage(updatedPage);
+      setSearchParams({ page: updatedPage.toString() });
       afterDisplay();
-      window.history.pushState(null, "", `?page=${updatedPage}`);
     }
   };
 
-  const missingMBDataAsListen = missingData.map((data) => {
-    return {
-      listened_at: new Date(data.listened_at).getTime() / 1000,
-      user_name: user.name,
-      track_metadata: {
-        artist_name: data.artist_name,
-        track_name: data.recording_name,
-        release_name: data?.release_name,
-        additional_info: {
-          recording_msid: data.recording_msid,
-        },
-      },
-    };
-  });
-
   // Effects
   React.useEffect(() => {
-    window.history.replaceState(null, "", `?page=${currPage}`);
-  }, []);
+    // Only run once
+    setSearchParams({ page: currPage.toString() });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [setSearchParams]);
```

Comment:
Worse than that, the endpoint was called when changing page, which does not make sensein this context since we load all the data we need the first time, and only split it visually for pagination.

Had to move away from RouteLoader and instead use useQuery in 5a8773ecd